spec/solver_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rspec'
+require_relative '../solver'
+
+RSpec.describe Solver do
+  describe '#Factorial Test' do
+    it 'should return the factorial of a number' do
+      solver = Solver.new
+      expect(solver.factorial(5)).to eq(120)
+    end
```

Comment:
- [ ] The #factorial method should have a test that handles the negative value which raises an exception as an expectation. Example is 
` #spec/resolver_spec.rb
     ...
     describe Resolver do
         describe "#factorial" do
             it "raises exception for negative number" do
             resolver = Resolver.new
             expect { resolver.factorial(-5) }.to raise_exception
             end
         end
     end
`